function-aws-api-proxy-test/src/main/java/io/micronaut/function/aws/proxy/test/HttpServerEmbeddedServer.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.function.aws.proxy.test;
+
+import com.sun.net.httpserver.HttpServer;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.http.server.HttpServerConfiguration;
+import io.micronaut.http.server.exceptions.HttpServerException;
+import io.micronaut.http.server.exceptions.ServerStartupException;
+import io.micronaut.runtime.ApplicationConfiguration;
+import io.micronaut.runtime.server.EmbeddedServer;
+
+import java.net.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Internal
+class HttpServerEmbeddedServer implements EmbeddedServer {
```

Comment:
@graemerocher @yawkat I would like to move this class to core. What do you think? That way I could use it in every cloud module (aws, gcp, azure).

Reply:
+1 to putting it into servlet